timesketch/api/v1/resources/sigma.py

```diff
@@ -92,10 +120,253 @@ def get(self, rule_uuid):
+    def delete(self, rule_uuid):
+        """Handles DELETE request to the resource.
+        Args:
+            rule_uuid: uuid of the rule
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def put(self, rule_uuid):
+        """Handles update request to Sigma rules
+        Args:
+            rule_uuid: uuid of the rule
+        Returns:
+            The updated sigma object in JSON (instance of
+            flask.wrappers.Response)
+        """
+
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e: # pylint: disable=broad-except
+            logger.error(
+                "Unable to get the Sigma rule",
+                exc_info=True,
+            )
+            abort(HTTP_STATUS_CODE_NOT_FOUND, f"ValueError {e}")
+
+        form = request.json
+        if not form:
+            form = request.data
+        if not form.validate_on_submit():
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Unable to validate form data.")
+
+        abort(HTTP_STATUS_CODE_NOT_FOUND, "Method not implemented yet")
+
+        # TODO(jaegeral): complete this method
+        rule_yaml = form.get("rule_yaml", "")
+        parsed_rule = ts_sigma_lib.get_sigma_rule_by_text(rule_yaml)
+
+        logger.debug(rule_yaml + parsed_rule + rule)
+
+
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        rule_yaml = form.get("rule_yaml", "")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule_yaml supplied.",
+            )
+
+        parsed_rule = ts_sigma_lib.get_sigma_rule_by_text(rule_yaml)
+
+        if not rule_uuid:
+            rule_uuid = parsed_rule.get("id")
+
+        if not isinstance(rule_yaml, str):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, "rule_yaml needs to be a string."
+            )
+
+        # Query rules to see if it already exist
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+        if rule:
+            if rule.rule_uuid == parsed_rule.get("rule_uuid"):
+                abort(HTTP_STATUS_CODE_CONFLICT, "Rule already exist")
+
+        sigma_rule = SigmaRule.get_or_create(
+            rule_yaml=form.get("rule_yaml", ""),
+            description = parsed_rule.get("description", ""),
+            title = parsed_rule.get("title", ""),
+            user = current_user
+            )
+
+        sigma_rule.query_string = parsed_rule.get("query_string", "")
```

Comment:
`sigma_rule.query_string`: Just to be sure, since I do not see this field existing in the SigmaRule class. Does it need to be defined there?

Reply:
yes it was `es_query` before and we decided to use the opportunity to rename it to a more general term